src/main/java/com/redhat/jenkins/plugins/ci/threads/TriggerThreadTransientActionFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.redhat.jenkins.plugins.ci.threads;
+
+import hudson.Extension;
+import hudson.model.Job;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import jenkins.model.TransientActionFactory;
+import jenkins.model.ParameterizedJobMixIn;
+
+import com.redhat.jenkins.plugins.ci.CIBuildTrigger;
+
+@Extension
+public class TriggerThreadTransientActionFactory extends TransientActionFactory {
```

Comment:
Should extend `TransientActionFactory<Job<?, ?>>` (I am surprised javac does not scream for raw type usage here). It will get the `target` argument typed correctly in `createFor`.

Reply:
I took this as far as I could as we discussed.  I had to use <Job> everywhere. There are still some rawtypes complaints but I'm not sure how or if they can be resolved.